frappe/core/doctype/user/user.js

```diff
@@ -340,6 +341,14 @@ frappe.ui.form.on("User", {
 			frappe.ui.toolbar.clear_cache();
 		}
 	},
+	setup_time_zone_autocomplete: function (frm) {
```

Comment:
Quite hacky. `Autocomplete` is supposed to drop-in replacement for select. 

The API mismatch needs to be handled, but it should work without too many changes.